client-runtime/src/main/java/com/microsoft/rest/v2/http/HttpRequest.java

```diff
@@ -18,6 +19,7 @@
     private String callerMethod;
     private HttpMethod httpMethod;
     private URL url;
+    private HttpResponseDecoder responseDecoder;
```

Comment:
Is there any reason a `responseDecoder` would change? Can we make this a constructor parameter and set it to final?

Reply:
Yeah, I've been thinking `final` might be best here--just have to deal with the churn in all the places where an HttpRequest was created.